modules/ROOT/pages/secure-storage.adoc

```diff
@@ -0,0 +1,110 @@
+
+=== Before you Begin
+
+* Ensure that all other instances of Anypoint Studio and any other Eclipse-based application are not running. +
+Because the secure storage file is shared across all running instances and the master password is cached in each instance's memory, updating your master password causes other opened instances to still write the credentials using the old master password. Closing all other instances ensures you avoid the <<no-password-provided-exception,No Password Provided Exception>>.
+
+To re-generate your master password:
+
+. Navigate to *Preferences* > *General* > *Security* > *Secure Storage*.
+. Select *Change Password...*:
++
+image::secure-storage-change-password.png[]
+. Studio prompts you to create a password hint for this new password. +
+Select *Yes* to generate a new password hint:
++
+image::new-password-created.png[]
++
+[CAUTION]
+--
+If you don't supply new password hints, then the former ones persist, which reference the former password. You cannot recover this new password using the previous password hints.
+--
+. Studio prompts you that the password has been successfully updated.
+. Select *Apply and Close*.
+
+
+== Customize the Location of the Secure Storage and Password Files
+
+To change the location of the password and secure storage files:
+
+. Locate your `AnypointStudio.ini` file:
+* In Windows:
+.. Go to your Anypoint Studio's installation directory.
+* In macOS:
+.. Go to your `/Applications` folder, right-click the `AnypointStudio` package and select *Show Package Contents*.
+.. Go to `Contents` > `Eclipse`, and open the `AnypointStudio.ini` file.
+* In Linux:
+.. Go to `/usr/lib/anypointstudio/anypointstudio.ini`.
+. Add the following options together to bypass the password provider mechanism:
++
+[source]
+--
+-eclipse.keyring <file_path> //<1>
+-eclipse.password <file_path> //<2>
+--
+<1> Configures the location of the Secure Storage file.
+<2> Configures the location of the password file. Studio uses this file's content as the master password.
+
+[CAUTION]
+You can only decrypt your credentials with the same password file used to encrypt them.
+
+
+
+== Troubleshooting the Secure Storage
+
+[[no-password-provided-exception]]
+=== No Password Provided
+
+Exception: `org.eclipse.equinox.security.storage.StorageException: No password provided`
+
+Studio throws this error when it loads and caches the master password for the provider and reads a password in the secure storage that cannot decrypt.
+
+This happens when one instance of Anypoint Studio changes the master password while another instance of Studio saves a new credential using a cached master password.
+
+
+=== Given Final Block Not Properly Padded
+
+`org.eclipse.equinox.security.storage.StorageException: Given final block not properly padded`.
+
+Studio throws this error when it attempts to use a bad key during decryption.
```

Comment:
Was on purpose to not add more information about this error like the previous one?

This line is base on the actual full error message that is thrown (which on the title was shortened here). It could confuse the user about the mention of "key" which is an internal name and even it is not the same I think it is better to write it as "master password".

Maybe I would explain better why it is produced and when this could happen.

Reply:
You are right. I didn't notice that I didn't complete this description. 

I just pushed an update. 79bf1da